src/test/java/it/cnr/isti/hpc/wikipedia/article/ArticleTest.java

```diff
@@ -124,13 +124,39 @@ public void testExtractingLinksOtherLang(){
 	}
 
 	/*
+	* Test for inline ref parsing.
+	* */
+	@Test
+	public void testReferences() throws IOException {
+		ArticleParser parser = new ArticleParser(Language.EN);
+		Article a = new Article();
+		String mediawiki = IOUtils.getFileAsUTF8String("./src/test/resources/misc/anarchism.xml");
+		parser.parse(a, mediawiki);
+
+		// check that the first paragraph of anarchism has 10 refs detected.
+		List<Ref> refs = a.getParagraphsWithLinks().get(0).getRefs();
+		assertEquals(10, refs.size());
+
+		// Check first and last detected references inside values
+		Ref fistRef = refs.get(0);
+		assertEquals(157, fistRef.getStart());
+		assertEquals(435, fistRef.getEnd());
+		assertNotEquals(-1, fistRef.getText().indexOf("a social philosophy that rejects authoritarian government and maintains"));
+
+		Ref lastRef = refs.get(9);
+		assertEquals(2529, lastRef.getStart());
+		assertEquals(2713, lastRef.getEnd());
+		assertNotEquals(-1, lastRef.getText().indexOf("The Concise Oxford Dictionary of Politics. Ed. Iain McLean and Alistair McMillan"));
+	}
```

Comment:
👍 for the test, but it would be more unit-test-like to make the `extractInlineReferences` method protected and then test it directly.

Some more fun things to test:
* `extractInlineReferences(null)`
* `extractInlineReferences("")`
* `extractInlineReferences("  ")`